Model/Ui/AdyenApplePayConfigProvider.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2017 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Model\Ui;
+
+use Magento\Checkout\Model\ConfigProviderInterface;
+use Magento\Payment\Helper\Data as PaymentHelper;
+
+class AdyenApplePayConfigProvider implements ConfigProviderInterface
+{
+
+    const CODE = 'adyen_apple_pay';
+
+    /**
+     * @var PaymentHelper
+     */
+    protected $_paymentHelper;
+
+    /**
+     * @var \Adyen\Payment\Helper\Data
+     */
+    protected $_adyenHelper;
+
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    protected $_urlBuilder;
+
+    /**
+     * Request object
+     *
+     * @var \Magento\Framework\App\RequestInterface
+     */
+    protected $_request;
+
+    /**
+     * AdyenApplePayConfigProvider constructor.
+     *
+     * @param PaymentHelper $paymentHelper
+     * @param \Adyen\Payment\Helper\Data $adyenHelper
+     * @param \Magento\Framework\App\RequestInterface $request
+     * @param \Magento\Framework\UrlInterface $urlBuilder
+     */
+    public function __construct(
+        PaymentHelper $paymentHelper,
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Magento\Framework\App\RequestInterface $request,
+        \Magento\Framework\UrlInterface $urlBuilder
+    ) {
+        $this->_paymentHelper = $paymentHelper;
+        $this->_adyenHelper = $adyenHelper;
+        $this->_request = $request;
+        $this->_urlBuilder = $urlBuilder;
+    }
+
+    /**
+     * Retrieve assoc array of checkout configuration
+     *
+     * @return array
+     */
+    public function getConfig()
+    {
+        // set to active
+        return [
+            'payment' => [
+                self::CODE => [
+                    'isActive' => true,
+                    'redirectUrl' => $this->_urlBuilder->getUrl(
+                        'checkout/onepage/success/', ['_secure' => $this->_getRequest()->isSecure()]),
```

Comment:
do we need the utm_nooverride as well?

Reply:
This is to exclude referral urls, this is coming from an internal page so no issues there.